src/layout/size_rules.rs

```diff
@@ -437,7 +437,6 @@ impl SizeRules {
             let target = target - margin_sum;
 
             if sum == target {
-                return;
```

Comment:
You tell me whether the method's flow is clear enough. May be okay, but definitely over-sized.